core/src/main/scala/org/broadinstitute/dsde/rawls/jobexec/SubmissionMonitorActor.scala

```diff
@@ -316,8 +316,9 @@ trait SubmissionMonitor extends FutureSupport with LazyLogging with RawlsInstrum
         for {
           costBreakdown <- executionServiceCluster.getCost(workflowRec, petUser)
           updatedWorkflowRec <-
-            // TODO CORE-217: if workflow is already in Aborted or Aborting, don't try to re-abort it
-            if (costBreakdown.cost > costCap) {
+            if (
+              costBreakdown.cost > costCap && costBreakdown.status != WorkflowStatuses.Aborted.toString && costBreakdown.status != WorkflowStatuses.Aborting.toString
+            ) {
```

Comment:
Should this also check `Failed` and `Succeeded` statuses? If the workflow has completed by the time we get here, we might want to just leave it alone. (I know this wasn't part of the original ticket)

Reply:
sure why not

Reply:
if we use the enum, we could implement an `isAbortable` function on it then this line could be
```
costBreakdown.cost > costCap && costBreakdown.status.isAbortable
```

Reply:
Sure but it seems like a bit of yak-shaving

Reply:
What about checking if it is one of the `abortableStatuses`? https://github.com/broadinstitute/rawls/blob/88dbae0eb6d29464d3be9d6d9a5dd01c6b953a8e/model/src/main/scala/org/broadinstitute/dsde/rawls/model/ExecutionModel.scala#L591